### PR TITLE
Don't call helm Get to pick up the latest version when there is only …

### PR DESCRIPTION
**Pull request by xiangjingli**
…one version for the given chart

* [X] I have taken backward compatibility into consideration.

https://issues.redhat.com/browse/ACM-7031?focusedId=24072452&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24072452

### CODE REVIEW

**Comment by xiangjingli** [comment]
@mikeshng   It turns out the helm Get fails to figure the latest version for the `3.0.0-stable` semVer pattern.

I feel the PR may be a easy fix for covering most cases - When specifying the version pattern `3.0.0-stable`, users may mainly want to fetch the exact version of the chart, So unlikely they will use this pattern`>3.0.0-stable`. Then only one version of the chart is fetched. Thus we won't call helm Get to figure out the latest version

**Review by mikeshng — APPROVED** [review]
/approve

/lgtm

/unhold

Thanks for the fix @xiangjingli I agree this fix is safe because it relaxes the restriction which fits in our maint mode only policy.

**Comment by xiangjingli** [comment]
@mikeshng  I realized there is an error in the previous fix due to my misunderstanding of the indexfile.entries
```
indexFile.Entries map[string]ChartVersions
```
Each map index is a chart name, and it could include multiple versions. So checking if there is only one version for the chart should be in each indexFile.Entries

File: pkg/utils/helmrepo.go [hunk 1]
```diff
@@ -538,7 +538,15 @@ func FilterCharts(sub *appv1.Subscription, indexFile *repo.IndexFile) error {
 func takeLatestVersion(indexFile *repo.IndexFile) (err error) {
 	indexFile.SortEntries()
 
-	for k := range indexFile.Entries {
+	// Don't call helm Get to pick up the latest version when there is only one version for the given chart. This is to avoid the error
+	// If there is only one version for the chart and it is defined as `3.0.0-stable`,
+	// the helm Get function fails to identify the version pattern with error
+	for k, chartVersions := range indexFile.Entries {
+		if len(chartVersions) == 1 {
+			klog.Infof("only one version is found, no need to helm Get latest version, chart name: %v, version: %v", k, chartVersions[0].Version)
```
Comment (mikeshng) [inline pkg/utils/helmrepo.go:546]:
```suggestion
			klog.V(2).Infof("only one version is found, no need to helm Get latest version, chart name: %v, version: %v", k, chartVersions[0].Version)
```

Let's avoid changing the default behaviour too much by suppressing this logging a bit.

**Pull request merged**